kaldb/src/test/java/com/slack/kaldb/metadata/service/ServicePartitionMetadataTest.java

```diff
@@ -79,4 +122,113 @@ public void testValidServicePartitionMetadata() {
         .isThrownBy(
             () -> new ServicePartitionMetadata(start.toEpochMilli(), end.toEpochMilli(), null));
   }
+
+  @Test
+  public void testOneServiceOnePartition() {
+    final String name = "testService";
+    final String owner = "serviceOwner";
+    final long throughputBytes = 1000;
+    final ServicePartitionMetadata partition = new ServicePartitionMetadata(100, 200, List.of("1"));
+
+    ServiceMetadata serviceMetadata =
+        new ServiceMetadata(name, owner, throughputBytes, List.of(partition));
+
+    serviceMetadataStore.createSync(serviceMetadata);
+    await().until(() -> serviceMetadataStore.listSync().size() == 1);
+
+    List<ServicePartitionMetadata> partitionMetadata =
+        findPartitionsToQuery(serviceMetadataStore, 101, 199, name);
+    assertThat(partitionMetadata.size()).isEqualTo(1);
+    assertThat(partitionMetadata.get(0).partitions.size()).isEqualTo(1);
+
+    partitionMetadata = findPartitionsToQuery(serviceMetadataStore, 1, 150, name);
+    assertThat(partitionMetadata.size()).isEqualTo(1);
+    assertThat(partitionMetadata.get(0).partitions.size()).isEqualTo(1);
+
+    partitionMetadata = findPartitionsToQuery(serviceMetadataStore, 1, 250, name);
+    assertThat(partitionMetadata.size()).isEqualTo(1);
+    assertThat(partitionMetadata.get(0).partitions.size()).isEqualTo(1);
+
+    partitionMetadata = findPartitionsToQuery(serviceMetadataStore, 200, 250, name);
+    assertThat(partitionMetadata.size()).isEqualTo(1);
+    assertThat(partitionMetadata.get(0).partitions.size()).isEqualTo(1);
+
+    partitionMetadata = findPartitionsToQuery(serviceMetadataStore, 201, 250, name);
+    assertThat(partitionMetadata.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void testOneServiceMultipleWindows() {
+    final String name = "testService";
+    final String owner = "serviceOwner";
+    final long throughputBytes = 1000;
+    final ServicePartitionMetadata partition1 =
+        new ServicePartitionMetadata(100, 200, List.of("1"));
+
+    final ServicePartitionMetadata partition2 =
+        new ServicePartitionMetadata(201, 300, List.of("2", "3"));
+
+    ServiceMetadata serviceMetadata =
+        new ServiceMetadata(name, owner, throughputBytes, List.of(partition1, partition2));
+
+    serviceMetadataStore.createSync(serviceMetadata);
+    await().until(() -> serviceMetadataStore.listSync().size() == 1);
+
+    List<ServicePartitionMetadata> partitionMetadata =
+        findPartitionsToQuery(serviceMetadataStore, 101, 199, name);
```

Comment:
Same as above, a few comments explaining what each of these blocks is attempting to prove.